packages/core/useAsyncState/index.ts

```diff
@@ -61,20 +61,33 @@ export interface AsyncStateOptions {
  * @param initialState    The initial state, used until the first evaluation finishes
  * @param options
  */
+
+export function useAsyncState<T>(
+  promise: Promise<T> | ((...args: any[]) => Promise<T>),
+  initialState: T,
+  options: Omit<AsyncStateOptions, 'shallow'> & { shallow: false }): UseAsyncStateReturn<T, false>
```

Comment:
Pass the true/false as a generic of `AsyncStateOptions`

Reply:
Thanks a lot for your advice